packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -230,74 +213,127 @@ class _TreemapState extends State<Treemap> {
-        isHorizontalRectangle ? y + pivotBestHeight : y + list1Height;
-    final list2Cells = buildTreemap(
-      children: list2,
-      width: list2Width,
-      height: list2Height,
-      x: list2XCoord,
-      y: list2YCoord,
-    );
+    final list2XCoord = isHorizontalRectangle ? list1Width : 0.0;
+    final list2YCoord = isHorizontalRectangle ? pivotBestHeight : list1Height;
+    if (list2.isNotEmpty) {
+      positionedTreemaps.add(
+        Positioned(
+          left: list2XCoord,
+          top: list2YCoord,
+          width: list2Width,
+          height: list2Height,
+          child: NewTreemap.fromNodes(
+            nodes: list2,
+            levelsVisible: widget.levelsVisible,
+            onRootChangedCallback: widget.onRootChangedCallback,
+          ),
+        ),
+      );
+    }
 
     // Construct pivot cell.
-    final pivotXCoord = isHorizontalRectangle ? x + list1Width : x + list2Width;
-    final pivotYCorrd = isHorizontalRectangle ? y : y + list1Height;
-    final pivotCell = TreemapCell(
-      key: UniqueKey(),
-      width: pivotBestWidth,
-      height: pivotBestHeight,
-      x: pivotXCoord,
-      y: pivotYCorrd,
-      node: pivotDataReference,
-      onTap: () => cellOnTap(rootNode.childrenMap[pivotDataReference.name]),
-      levelsVisible: widget.levelsVisible - 1,
+    final pivotXCoord = isHorizontalRectangle ? list1Width : list2Width;
+    final pivotYCoord = isHorizontalRectangle ? 0.0 : list1Height;
+
+    positionedTreemaps.add(
+      Positioned(
+        left: pivotXCoord,
+        top: pivotYCoord,
+        width: pivotBestWidth,
+        height: pivotBestHeight,
+        child: NewTreemap.fromRoot(
+          rootNode: pivotDataReference,
+          levelsVisible: widget.levelsVisible - 1,
+          onRootChangedCallback: widget.onRootChangedCallback,
+        ),
+      ),
     );
 
     // Construct list 3 sub-treemap.
-    final list3Ratio = list3ByteSize / totalSize;
+    final list3Ratio = list3ByteSize / totalByteSize;
     final list3Width = isHorizontalRectangle ? list3Ratio * width : width;
     final list3Height = isHorizontalRectangle ? height : list3Ratio * height;
     final list3XCoord =
-        isHorizontalRectangle ? x + list1Width + pivotBestWidth : x;
+        isHorizontalRectangle ? list1Width + pivotBestWidth : 0.0;
     final list3YCoord =
-        isHorizontalRectangle ? y : y + list1Height + pivotBestHeight;
-    final list3Cells = buildTreemap(
-      children: list3,
-      width: list3Width,
-      height: list3Height,
-      x: list3XCoord,
-      y: list3YCoord,
+        isHorizontalRectangle ? 0.0 : list1Height + pivotBestHeight;
+
+    if (list3.isNotEmpty) {
+      positionedTreemaps.add(
+        Positioned(
+          left: list3XCoord,
+          top: list3YCoord,
+          width: list3Width,
+          height: list3Height,
+          child: NewTreemap.fromNodes(
+            nodes: list3,
+            levelsVisible: widget.levelsVisible,
```

Comment:
comment why this isn't levelsVisible - 1 as that is slightly surprising unless a user understands the algorithm.

Reply:
I added some documentation for levelsVisible instead, since there are multiple instances where `levelsVisible - 1` is not passed in. Could you take a look and see if that's helpful?